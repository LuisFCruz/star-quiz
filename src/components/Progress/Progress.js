import CircularProgress from '@material-ui/core/CircularProgress';
import React, { Component } from 'react';

const styles = {
  mask: {
    position: 'fixed',
    width: '100%',
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    background: '#fff',
    zIndex: '1000',
    flexDirection: 'column'
  }
};

export class Progress extends Component {
  render() {

    return (
      <div style={styles.mask}>
        <CircularProgress size={100}/>
        <p>Preparando jogo...</p>
      </div>
    );
  }
}

export default Progress;
