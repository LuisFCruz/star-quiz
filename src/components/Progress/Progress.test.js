import React from 'react';
import { shallow } from 'enzyme';

import { Progress } from './Progress';

describe('<Progress />', () => {
  const props = {};
  const wrapper = shallow(<Progress {...props} />);

  test('smoke test', () => {
      expect(Progress).toBeDefined();
      expect(wrapper.instance()).toBeInstanceOf(Progress);
  });

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
