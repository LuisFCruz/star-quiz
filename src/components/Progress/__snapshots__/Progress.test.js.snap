// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Progress />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(CircularProgress)
          size={100}
        />,
        <p>
          Preparando jogo...
        </p>,
      ],
      "style": Object {
        "alignItems": "center",
        "background": "#fff",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "position": "fixed",
        "width": "100%",
        "zIndex": "1000",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "size": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Preparando jogo...",
        },
        "ref": null,
        "rendered": "Preparando jogo...",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(CircularProgress)
            size={100}
          />,
          <p>
            Preparando jogo...
          </p>,
        ],
        "style": Object {
          "alignItems": "center",
          "background": "#fff",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "center",
          "position": "fixed",
          "width": "100%",
          "zIndex": "1000",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "size": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Preparando jogo...",
          },
          "ref": null,
          "rendered": "Preparando jogo...",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
